src/rascal_utility.cc

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file   rascal_utility.cc
+ *
+ * Copyright  2019 COSMO (EPFL), LAMMM (EPFL)
+ *
+ * Rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * Rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include <cstdlib>
+#include <fstream>
+#include <iterator>
+
+#include "rascal_utility.hh"
+
+#ifdef __GNUG__
+#include <cxxabi.h>
+#endif
+
+std::string rascal::internal::type_name_demangled(const char * name) {
+#ifdef __GNUG__
```

Comment:
Is this flag also present with clang ?

Reply:
Yes, clang pretends to be GCC 4.2. Intel compiler also defines this (see https://stackoverflow.com/q/38499462)